src/VideoStreaming/VideoStreaming.cc

```diff
@@ -170,8 +164,17 @@ void initializeVideoStreaming(int &argc, char* argv[], char* logpath, char* debu
     Q_UNUSED(logpath);
     Q_UNUSED(debuglevel);
 #endif
-    qmlRegisterType<VideoItem>              ("QGroundControl.QgcQtGStreamer", 1, 0, "VideoItem");
-    qmlRegisterUncreatableType<VideoSurface>("QGroundControl.QgcQtGStreamer", 1, 0, "VideoSurface", QStringLiteral("VideoSurface from QML is not supported"));
+    /* the plugin must be loaded before loading the qml file to register the
+     * GstGLVideoItem qml item
+     * FIXME Add a QQmlExtensionPlugin into qmlglsink to register GstGLVideoItem
+     * with the QML engine, then remove this */
+    GstElement *sink = gst_element_factory_make ("qmlglsink", NULL);
+    if (sink != NULL) {
```

Comment:
nullptr, we are in c++

Reply:
> I don't think we need to download the whole gstreamer code for QGC, but considering that it's a subtree I'm mostly ok with it. 

There is no 'whole gstreamer' code. What is included is called gst-plugins-good and it is minimal unit which could be referenced to get qmlglsink into QGC.

> I'd prefer to use the system installed libraries if they exist and pull the submodules if they don't.

System installed libraries are supported and have highest priority - we use our own qmlglsink build only if we are unable to load user-provided library.

Moreover, using system installed libraries on Android and iOS is not possible, at all (rooting and firmware tweaking is out of scope).

IMO it is better to concentrate on common case and common case is 'static linking for all platforms and honor user-provided build, if it exists. I'm still asking myself if loading user-provided build is good option 'by default' since it is unlikely for average user to build qmlglsink which will target exactly the same Qt as QGC uses (qmlglsink uses private Qt headers - it will fail/crash/etc if Qt versions do no match). Perhaps QGC should try user-provided build only if some kind of 'developer mode' is turned on. It is probably less harmful option for everyone.

> the result is that "+687,585 −6,038 lines" does not really feel like a cleanup. but I can live with that.

The reason of that PR is to lower maintenance (human) efforts but not to get lowest line count. Most of these +687,585 are maintained by gstreamer team and this is point where everyone earns - I'm submitting qmlglsink patches upstream to fix issues which are relevant for QGC and gstreamer team does rest (maintaining other parts of qmlglsink).

Having +100 -6000 lines is way to nowhere if you have to work on these +100 whole time. But I agree +100 -6000 sounds impressive on first sight while "+687,585 −6,038 lines" has more weight on second and following sights.

NULL - this was fixed on December 12th, 2019 (see https://github.com/mavlink/qgroundcontrol/commit/2cfeca0c1ccbd094783a8db4f7d4af4e29126f7f)